src/persistence/SubjectExecutor.ts

```diff
@@ -607,8 +608,12 @@ export class SubjectExecutor {
                 } else { // in this case identifier is just conditions object to update by
                     softDeleteQueryBuilder.where(subject.identifier);
                 }
-
                 updateResult = await softDeleteQueryBuilder.execute();
+                // Move throw all the relation of the subject
+                for (const relation of subject.metadata.relations) {
+                    // Call recursive function that get the parents primary keys that in used on the inverse side in all one to many relations
+                    await this.executeSoftRemoveRecursive(relation, [Reflect.get(subject.identifier, subject.metadata.primaryColumns[0].propertyName)]);
```

Comment:
not really clear why `Reflect.get` is used here. Decorators isn't the only source of entity definitions and some people define entities using entity schemas.